core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -190,6 +206,88 @@ newtype Shelley (depth :: Depth) key = Shelley
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.root_xsk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xprvFromBytes dec)
+instance FromBech32 (Shelley 'RootK XPub) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.root_xvk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xpubFromBytes dec)
+
+instance ToBech32 (Shelley 'AccountK XPrv) where
+    bech32 (Shelley k) = bech32With CIP5.acct_xsk $ xprvToBytes k
+instance ToBech32 (Shelley 'AccountK XPub) where
+    bech32 (Shelley k) = bech32With CIP5.acct_xvk $ xpubToBytes k
+instance FromBech32 (Shelley 'AccountK XPrv) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.acct_xsk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xprvFromBytes dec)
+instance FromBech32 (Shelley 'AccountK XPub) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.acct_xvk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xpubFromBytes dec)
+
+instance ToBech32 (Shelley 'PaymentK XPrv) where
+    bech32 (Shelley k) = bech32With CIP5.addr_xsk $ xprvToBytes k
+instance ToBech32 (Shelley 'PaymentK XPub) where
+    bech32 (Shelley k) = bech32With CIP5.addr_xvk $ xpubToBytes k
+instance FromBech32 (Shelley 'PaymentK XPrv) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.addr_xsk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xprvFromBytes dec)
+instance FromBech32 (Shelley 'PaymentK XPub) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.addr_xvk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xpubFromBytes dec)
+
+instance ToBech32 (Shelley 'DelegationK XPrv) where
+    bech32 (Shelley k) = bech32With CIP5.stake_xsk $ xprvToBytes k
+instance ToBech32 (Shelley 'DelegationK XPub) where
+    bech32 (Shelley k) = bech32With CIP5.stake_xvk $ xpubToBytes k
+instance FromBech32 (Shelley 'DelegationK XPrv) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.stake_xsk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xprvFromBytes dec)
+instance FromBech32 (Shelley 'DelegationK XPub) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.stake_xvk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xpubFromBytes dec)
+
+instance ToBech32 (Shelley 'ScriptK XPrv) where
+    bech32 (Shelley k) = bech32With CIP5.script_xsk $ xprvToBytes k
+instance ToBech32 (Shelley 'ScriptK XPub) where
+    bech32 (Shelley k) = bech32With CIP5.script_xvk $ xpubToBytes k
+instance FromBech32 (Shelley 'ScriptK XPrv) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.script_xsk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xprvFromBytes dec)
+instance FromBech32 (Shelley 'ScriptK XPub) where
+    fromBech32 b = do
+        dec <- fromBech32With CIP5.script_xvk id b
+        Shelley <$> maybeToEither "Invalid Key Length" (xpubFromBytes dec)
+
+newtype WithoutChain key = WithoutChain key
```

Comment:
I'd suggest: `WithoutChainCode`, words are cheap, so let's use proper qualified when we can :+1: